cowprotocol/cowswap
# Fix zero fee check to show free text

# Summary

Fixes #2015 

Our implementation for checking whether or not there is a zero / non-zero fee was incorrect as it was looking for falsy strings. Also, showing fiat equivalent had no checks regarding whether it should be shown, as when it is free, it is not necessary.

I've added a utility to check for a string amount being a valid non-zero amount, and updated the logic to show free correctly as shown down below.

<img width="518" alt="Screenshot 2023-02-17 at 03 37 49" src="https://user-images.githubusercontent.com/5172699/219535936-caccf335-0e1d-447a-ae79-a1bf49f211e2.png">
<img width="527" alt="Screenshot 2023-02-17 at 03 37 28" src="https://user-images.githubusercontent.com/5172699/219535940-d3bf864b-ba97-4d7b-8db8-faafac3071a0.png">


  # To Test

Free case:
1. Using https://www.impersonator.xyz/ connect to this account 0xDe1E8EedD42CD284a3052d987502d0B24c633FEE (this is a liquidity provider)
2. Open Swap page and specify a token pair, amount to trade
3. Check the Fee field
4. Expect;
- [] Free should be shown without any numbers.

Non-free case:
1. Using any non liquidity provider wallet connect to your account
2. Open swap page and specify a token pair that would have fees
3. Check the fee field
4. Expect;
- [] Fee should be shown with its fiat equivalent next to it.
